src/main/webapp/components/services/utils.js

```diff
@@ -56,6 +56,58 @@
           return false;
         }
         return true;
+      },
+
+      setValue: function setValue(object, path, value) {
+        var a = path.split('.');
+        var o = object;
+        for (var i = 0; i < a.length - 1; i++) {
+          var n = a[i];
+          if (n in o) {
+            o = o[n];
+          } else {
+            o[n] = {};
+            o = o[n];
+          }
+        }
+        o[a[a.length - 1]] = value;
+      },
+
+      getValue: function getValue(object, path) {
```

Comment:
Check out above, similar like setValue, but it gets value, and will not create any empty objects -> it will return undefined instead (instead of an error).